app/Model/CoEnrollmentAttribute.php

```diff
@@ -112,8 +112,42 @@ class CoEnrollmentAttribute extends AppModel {
       'rule' => array('numeric'),
       'required' => false,
       'allowEmpty' => true
+    ),
+    'hidden' => array(
+      'ruleName' => array(
+        'rule' => array('boolean'),
+        'required' => true // All attributes class have a hidden value sent even if they are not applicable
+      ),
+      'ruleName2' => array(
+        'rule' => array('validateHidden'),
+        'required' => false, 
+        'message' => 'A hidden field must have a default value'
+      )
     )
-  );  
+  );
+
+  /**
+   * Validate that the field is allowed to be hidden based on the attribute and default value
+   *
+   * @param  array Array of fields to validate
+   * @return boolean True if allowed, false if not
+   */
+  public function validateHidden($a) {
+    // Hidden attribute is only applicable to CO Person Role attributes (type 'r'), CO Person attributes (type 'g'),
+    // Organizational Identity attributes (type 'o'), or Extended Attributes (type 'x')
+    $attrCode = explode(':', $this->data['CoEnrollmentAttribute']['attribute']);
+    if($attrCode[0] != 'r' && $attrCode[0] != 'g' && $attrCode[0] != 'o' && $attrCode[0] != 'x') {
```

Comment:
Nit picky - but would in_array work here instead of stringing several checks together?

Reply:
@Jeanesily yep, I saw this somewhere else, but the `in_array` thing is way more readable.